src/lib/sample_metadata.rs

```diff
@@ -171,6 +171,29 @@ impl SampleMetadata {
         Ok(self)
     }
 
+    /// Returns a "semantic" stringified barcode where its read structure is explcitly delimited and
+    /// readable.
+    /// - For dual index barcodes where `self.index1` and `self.index2` are both
+    ///    `Some`, this returns the barcodes concatenated and delimited with `+`
+    /// - Otherwise, returns `self.raw_barcode` if it is not `None`
+    /// - Else, returns `self.barcode`
+    ///
+    /// **Note**: this expects that `self` is a valid SampleMetdata struct. i.e., a after
+    /// an update and sanitization with `update_with_and_set_demux_barcode()`, or is a sentinal
+    /// value where the barcode is all Ns.
+    pub fn get_semantic_barcode(&self) -> BString {
```

Comment:
@nh13 refactored this into `SampleMetadta` and added a docstring per your previous comment.

Reply:
Will want to use this in metrics.rs given you are creating the unmatched barcode there.  Sorry, you may have to revert this to being not a method on samplemetadata